BPFTRACE_KERNEL_SOURCE  environment variable

cneira opened the pull request:
Currently, bpftrace expects sources to be in:
```bash
/lib/modules/$(uname -r) 
```
To address this the environment variable BPFTRACE_KERNEL_SOURCE has been added to the code if the variable does not exist it will search the modules directory as before. 
Issue is referenced in https://github.com/iovisor/bpftrace/issues/169

brendangregg commented:
"bpftrace depends on kernel sources" -- it's kernel headers, right, not kernel sources? Saying people need kernel sources sounds like they need the entire kernel, but I believe it's just the kernel headers package.

cneira commented:
> "bpftrace depends on kernel sources" -- it's kernel headers, right, not kernel sources? Saying people need kernel sources sounds like they need the entire kernel, but I believe it's just the kernel headers package.

You are right, in my system (Debian stretch) the package linux-headers-4.9.0-8-common is installed but does not provide generated/autoconf.h, however, Ubuntu bionic does offer it https://packages.ubuntu.com/bionic-updates/amd64/linux-headers-4.15.0-32-generic/filelist.
My workaround was to set up BPFTRACE_KERNEL_SOURCE to the Linux kernel sources, but as you correctly state, bpftrace only needs kernel headers.
I'll fix that in the commit.
Thanks for reviewing this change.

mmarchini commented on src/clang_parser.cpp:
```diff
@@ -159,10 +159,25 @@ void ClangParser::parse(ast::Program *program, StructMap &structs)
+	std::cerr << "WARNING: ("
+		  << kernel_modules_dir << ") is not a valid dir" << std::endl;
+	std::cerr << "Use environment variable BPFTRACE_KERNEL_SOURCE to setup"
+		     " a valid directory for kernel sources." << std::endl;
```
s/kernel source/kernel headers/

cneira replied:
You are right, for consistency's sake BPFTRACE_KERNEL_HEADERS is the correct name.

mmarchini commented on README.md:
```diff
@@ -2,7 +2,14 @@
-To learn more about BPFtrace, see the [Reference Guide](docs/reference_guide.md) and [One-Liner Tutorial](docs/tutorial_one_liners.md).
+To learn more about BPFtrace, see the [Reference Guide](docs/reference_guide.md) and [One-Liner Tutorial](docs/tutorial_one_liners.md).  
+
+BPFTrace depends on the kernel headers which are searched for by default in :  
```
nit: remove extra space between `in` and `:`

cneira commented:
@mmarchini 
Good catch!. You are right, for consistency's sake BPFTRACE_KERNEL_HEADERS is the correct name. I have also addressed your comment on README.md. 
Thanks!

brendangregg commented:
ok, thanks!
